weave/util.py

```diff
@@ -160,3 +160,24 @@ def relpath_no_syscalls(
         common_length:
     ]
     return os.path.sep.join(relative_parts)
+
+def sample_rows(data: list, max_rows: int) -> list:
+    data_len = len(data)
+
+    if data_len <= max_rows:
+        return data
+    
+    if max_rows <= 0:
+        return []
+    if max_rows == 1:
+        return [data[0]]
+    if max_rows == 2:
+        return [data[0], data[-1]]
```

Comment:
Special-case logic to handle values of `max_rows < 3` is necessary because the 3-segment sampling logic below breaks in those cases.

Specifically, `split_size = max_rows // 3` cannot be `0` because then `end_split = data[-split_size:]` will be the entire `data` array.